src/components/ChannelSearch/utils.ts

```diff
@@ -0,0 +1,34 @@
+import type { Channel, UserResponse } from 'stream-chat';
+
+import type {
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+  DefaultEventType,
+  DefaultMessageType,
+  DefaultReactionType,
+  DefaultUserType,
+} from '../../types/types';
+
+export type ChannelOrUserResponse<
+  At extends DefaultAttachmentType = DefaultAttachmentType,
+  Ch extends DefaultChannelType = DefaultChannelType,
+  Co extends DefaultCommandType = DefaultCommandType,
+  Ev extends DefaultEventType = DefaultEventType,
+  Me extends DefaultMessageType = DefaultMessageType,
+  Re extends DefaultReactionType = DefaultReactionType,
+  Us extends DefaultUserType<Us> = DefaultUserType
+> = Channel<At, Ch, Co, Ev, Me, Re, Us> | UserResponse<Us>;
+
+export const isChannel = <
+  At extends DefaultAttachmentType = DefaultAttachmentType,
+  Ch extends DefaultChannelType = DefaultChannelType,
+  Co extends DefaultCommandType = DefaultCommandType,
+  Ev extends DefaultEventType = DefaultEventType,
+  Me extends DefaultMessageType = DefaultMessageType,
+  Re extends DefaultReactionType = DefaultReactionType,
+  Us extends DefaultUserType<Us> = DefaultUserType
+>(
+  channel: ChannelOrUserResponse<At, Ch, Co, Ev, Me, Re, Us>,
+): channel is Channel<At, Ch, Co, Ev, Me, Re, Us> =>
+  (channel as Channel<At, Ch, Co, Ev, Me, Re, Us>).cid != null;
```

Comment:
change these `channel` instances to `output` to match other type guards and since not always a channel being passed in